### PR TITLE
Add set_peer_timeout to NetworkedMultiplayerENet.

### LINKED ISSUE #20056: Expose ENet peer timeout settings
*Bugsquad edit:* This issue is about exposing http://enet.bespin.org/group__peer.html#gac48f35cdd39a89318a7b4fc19920b21b

----

So using the latest godot master branch. I'm working on a multiplayer game. Now sometimes I get about half an hour in to the game and there is a lag spike then I get a disconnect as a result.

It seems as if some packet exceeds its timeout, which causes a disconnect by the server. The client then locks up in PacketPeerUDPPosix::_poll while it waits for recvfrom (which is meant to be non-blocking? so how it can block there is beyond me). After about 30 seconds the client will continue and detect the disconnect. During this time, the game is totally locked up. I can't even get a signal to be emitted that the client is waiting for the server. This should be fixed.

More importantly, there should be a way to change the timeout settings for ENET. It seems a packet has about 1.6 seconds to be acknowledged as being received from the client, so that means its got about 0.8seconds to get tot he client, then 0.8seconds back at most. Any longer and then server will drop them.

Exposing a timeout would allow me to better handle low frame rates, playing over the internet. It would also allow me to easier to debug the network code - I would be able to set a break point and examine code and resume a multiplayer game without the fear of hitting a break point followed by a sudden disconnect.

### PR DESCRIPTION
**Pull request by Faless**
Closes #35349 (superseded). Added docs, error messages.

Fixes #20056 .
Probably worth a cherry-pick, should auto-merge.

### CODE REVIEW

File: modules/enet/doc_classes/NetworkedMultiplayerENet.xml [hunk 1]
```diff
@@ -124,6 +124,23 @@
...
+			</argument>
+			<argument index="1" name="timeout_limit" type="int">
+			</argument>
+			<argument index="2" name="timeout_min" type="int">
+			</argument>
+			<argument index="3" name="timeout_max" type="int">
+			</argument>
+			<description>
+				Sets the timeout parameters for a peer.	The timeout parameters control how and when a peer will timeout from a failure to acknowledge reliable traffic. Timeout values are expressed in milliseconds.
+
```
Comment (akien-mga) [inline modules/enet/doc_classes/NetworkedMultiplayerENet.xml:140]: Should not have empty lines (== empty paragraph).

Comment (Faless) [reply modules/enet/doc_classes/NetworkedMultiplayerENet.xml]: always forget this is not markdown :smirk_cat:

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Comment by MikeSchulze** [comment]
is this fix now included in the v3.x branch? you removed the cherrypick label

**Comment by akien-mga** [comment]
Yes, it was cherry-picked on March 1, so it's included in 3.2.4 RC 4 and later.

**Comment by MikeSchulze** [comment]
thanks ;) i downloade RC_8 and it works like a charm

**Comment by tavurth** [comment]
Was this removed from Godot 4.2?

**Comment by akien-mga** [comment]
I believe in 4.0+ this is `ENetPacketPeer::set_timeout`.

**Pull request merged**